### PR TITLE
Passing common data to all widgets

### PR DESCRIPTION
**Pull request by wafflepie**
I also changed the `name` config property to `widget` for consistency (only internal, no changes in public API).

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -8,5 +8,13 @@ module.exports = {
 				devDependencies: true,
 			},
 		],
+		'space-before-function-paren': [
```
Comment (tommmyy) [inline .eslintrc.js]: 👍

File: packages/react-union/src/scan.js [hunk 2]
```diff
@@ -69,7 +84,7 @@ const loadConfigs = (routes, descriptors) => {
  *											[{
  *												component,
  *												descriptor: {
- *													name,
+ *													widget,
```
Comment (tommmyy) [inline packages/react-union/src/scan.js]: are you sure you want that breaking change?

File: packages/react-union/API.md [hunk 3]
```diff
@@ -197,13 +197,13 @@ If you need to provide custom data to the instance of your widget, pass a valid
 </script>
 ```
 
-**Data Widget**
+**Passing common data to all widgets**
 
-If you only need to provide data to other widgets in your application, there is no need to render anything.
+If you need to provide common data to all widgets in your application, use the `data-union-common` attribute. This data will be recursively merged with the JSON content of each widget descriptor.
```
Comment (tommmyy) [inline packages/react-union/API.md]: Please, add info, that is encouraged to use just one common config at the view if the rendering engine cannot ensure the order of rendering.

File: packages/react-union/src/scan.js (hunk 2 above)
Comment (wafflepie) [reply packages/react-union/src/scan.js]: Yes, I think that at this stage of development we should encourage consistency as much as we can (keeping naming consistent across widget descriptors and the JavaScript API)

File: packages/react-union/API.md (hunk 3 above)
Comment (wafflepie) [reply packages/react-union/API.md]: Good point! Will do.

File: packages/react-union/src/scan.js (hunk 2 above)
Comment (wafflepie) [reply packages/react-union/src/scan.js]: In this case, the breaking change is negligible (and to be honest I did not even notice it before): we pass `configs` only to `onScanEnd`.

**Pull request merged**